subgraphs/notional-finance/src/mappings/notional.ts

```diff
@@ -0,0 +1,338 @@
+    event.params.maturity
+  ).notional;
+
+  // LendBorrow amounts (assetCash, fCash, USD)
+  let cTokenAmount = event.params.netAssetCash;
+  let fCashAmount = event.params.netfCash;
+  let amountUSD = bigIntToBigDecimal(cTokenAmount, token.decimals).times(
+    token.lastPriceUSD!
+  );
+
+  // we need absolute amounts for metrics; fCash is signed
+  let absAmountUSD: BigDecimal = amountUSD;
+  let absAmount: BigInt = cTokenAmount;
+  let absfCashAmount: BigInt = fCashAmount;
+  if (cTokenAmount < BIGINT_ZERO) {
+    absAmountUSD = amountUSD.neg();
+    absAmount = cTokenAmount.neg();
+  }
+  if (fCashAmount < BIGINT_ZERO) {
+    absfCashAmount = fCashAmount.neg();
+  }
+
+  // TODO: Need to deal with int and float formats for getting the rate
+  // let RATE_PRECISION: BigDecimal = BIGDECIMAL_ONE;
+  // let SECONDS_IN_YEAR: BigInt = BigInt.fromI32(60 * 60 * 24 * 365);
+  // let exchangeRate: BigDecimal = BIGDECIMAL_ZERO;
+  // exchangeRate = bigIntToBigDecimal(absfCashAmount.div(absAmount));
+
+  // let exchangeRateWithRatePrecision: BigDecimal = BIGDECIMAL_ZERO;
+  // exchangeRateWithRatePrecision = bigIntToBigDecimal(
+  //   bigIntToBigDecimal(absfCashAmount)
+  //     .times(RATE_PRECISION)
+  //     .div(absAmount)
+  // );
+
+  // let annualizedRate: BigDecimal = BIGDECIMAL_ZERO;
+  // let annualizedRateWithRatePrecision: BigInt = BIGINT_ZERO;
+  // let timeToMaturity: BigInt = maturity.minus(event.block.timestamp);
+  // annualizedRateWithRatePrecision = RATE_PRECISION;
+  // let whatever: BigInt = Math.log(
+  //   exchangeRateWithRatePrecision.div(RATE_PRECISION)
+  // )
+  //   .times(SECONDS_IN_YEAR)
+  //   .div(timeToMaturity)
+  //   .times(RATE_PRECISION);
+
+  // log.error(
+  //   " ---> currencyId: {}, absfCash: {}, absAssetCash: {}, exchangeRate: {}, exchangeRateWithRatePrecision: {}",
+  //   [
+  //     currencyId.toString(),
+  //     absfCashAmount.toString(),
+  //     absAmount.toString(),
+  //     exchangeRate.toString(),
+  //     exchangeRateWithRatePrecision.toString(),
+  //     // annualizedRate.toString(),
+  //     // annualizedRateWithRatePrecision.toString(),
+  //   ]
+  // );
+
+  // identify transaction type
+  // transactions of different user intention may call the same action type in notional smart contract design
+  // TODO: is there a way to identify if a transaction type is also liquidation event?
+  if (
+    fCashBeforeTransaction <= BIGINT_ZERO &&
+    fCashAfterTransaction < fCashBeforeTransaction
+  ) {
+    createBorrow(event, market, absfCashAmount, absAmount, absAmountUSD);
+  } else if (
+    // This means they withdrew their deposit and borrowed at the same time.
+    // No such events found but worth logging them.
+    // TODO: Methodology Question: A) Is this possible? B) How should this be treated? Currently, I'm treating this as borrow.
+    fCashBeforeTransaction > BIGINT_ZERO &&
+    fCashAfterTransaction < BIGINT_ZERO &&
+    fCashAfterTransaction < fCashBeforeTransaction
```

Comment:
Can you move these conditions to their own functions to make it clear what each combination means?

```
function isBorrow() {}
function isRepay() {}
...
```